components/micro-gateway-core/src/main/ballerina/src/gateway/utils/analytics_util.bal

```diff
@@ -73,6 +64,18 @@ function populateThrottleAnalyticsDTO(http:FilterContext context) returns (Throt
         eventDto.subscriber = END_USER_ANONYMOUS;
     }
 
+    APIConfiguration? apiConfiguration = apiConfigAnnotationMap[context.getServiceName()];
+    if (apiConfiguration is APIConfiguration) {
+        eventDto.apiVersion = apiConfiguration.apiVersion;
+        if (!stringutils:equalsIgnoreCase("", <string>apiConfiguration.publisher)
+                && stringutils:equalsIgnoreCase("", <string>eventDto.apiCreator)) {
+            eventDto.apiCreator = <string>apiConfiguration.publisher;
+        } else if (stringutils:equalsIgnoreCase("", <string>eventDto.apiCreator)) {
+            //sets API creator if x-wso2-owner extension not specified.
+            eventDto.apiCreator = UNKNOWN_VALUE;
+        }
+    }
+
```

Comment:
can we move this to separate function, used in two places

Reply:
It seems not possible since the DTO's are different in those two locations.